src/main/scala/singleton/ops/impl/GeneralMacros.scala

```diff
@@ -881,27 +900,64 @@ trait GeneralMacros {
     val (typedTree, tpe) = GetArgTree(argIdx, lhs)
     VerboseTraversal(s"@@extractFromArg@@\nTP: $tpe\nRAW: ${showRaw(tpe)}\nTree: $typedTree")
     TypeCalc(tpe) match {
-      case _ : CalcUnknown => CalcUnknown(tpe, Some(c.untypecheck(typedTree)))
+      case _ : CalcUnknown => CalcUnknown(tpe, Some(c.untypecheck(typedTree)), opIntercept = false)
       case t : CalcNLit => CalcNLit(t, typedTree)
       case t => t
     }
   }
   ///////////////////////////////////////////////////////////////////////////////////////////
 
+  ///////////////////////////////////////////////////////////////////////////////////////////
+  // OpInterept Result Caching
+  ///////////////////////////////////////////////////////////////////////////////////////////
+  def cacheOpInterceptResult[Out](implicit ev0: c.WeakTypeTag[Out]) : Tree  = {
+    val outTpe = weakTypeOf[Out]
+    val outCalc = TypeCalc(outTpe)
+    CalcCache.setOpInterceptCalc(outCalc)
+    q"new _root_.singleton.ops.OpIntercept.CacheResult[$outTpe]{}"
+  }
+  ///////////////////////////////////////////////////////////////////////////////////////////
+
   ///////////////////////////////////////////////////////////////////////////////////////////
   // Three operands (Generic)
   ///////////////////////////////////////////////////////////////////////////////////////////
   def materializeOpGen[F](implicit ev0: c.WeakTypeTag[F]): MaterializeOpAuxGen =
     new MaterializeOpAuxGen(weakTypeOf[F])
 
-  def opCalc(funcType : TypeSymbol, aCalc : => Calc, bCalc : => Calc, cCalc : => Calc) : Calc = {
+  def opCalc(opTpe : Option[Type], funcType : TypeSymbol, aCalc : => Calc, bCalc : => Calc, cCalc : => Calc) : Calc = {
     lazy val a = aCalc
     lazy val b = bCalc
     lazy val cArg = cCalc
     def unsupported() : Calc = {
-      (a, b) match {
-        case (aArg : CalcVal, bArg : CalcVal) => abort(s"Unsupported $funcType[$a, $b, $cArg]")
-        case _ => CalcUnknown(funcType.toType, None)
+      val cachedTpe = opTpe.get match {
+        case TypeRef(pre, sym, args) => c.internal.typeRef(pre, sym, List(funcType.toType, a.tpe, b.tpe, cArg.tpe))
+      }
+      //calling OpIntercept for the operation should cache the expected result if executed correctly
+      CalcCache.clearOpInterceptCalc()
+      val implicitlyTree = q"implicitly[_root_.singleton.ops.OpIntercept[$cachedTpe]]"
+      try {
+        c.typecheck(implicitlyTree, silent = false)
+        val cachedCalc = CalcCache.getOpInterceptCalc match {
+          case Some(calc) => calc
+          case None => abort("Missing a result cache for OpIntercept. Make sure you set `OpIntercept.CacheResult`")
+        }
+        CalcCache.clearOpInterceptCalc()
+        cachedCalc match {
+          case Left(t : CalcUnknown) =>
```

Comment:
this is totally nitpicking, but IIUC, the convention for `Either` is that the `Left` is the "error" condition, isn't it?

Reply:
You're right. I've rarely used it in this fashion, so forgot. I'll fix.